core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala

```diff
@@ -251,6 +255,69 @@ class SubmissionComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers
     }
   }
 
+  it should "count workflows by queue status" in withDefaultTestDatabase {
+    // Create some test submissions
+    val statusCounts = Map(WorkflowStatuses.Submitted -> 1, WorkflowStatuses.Running -> 10, WorkflowStatuses.Aborting -> 100)
+    withWorkspaceContext(testData.workspace) { ctx =>
+      statusCounts.flatMap { case (st, count) =>
+        for (_ <- 0 until count) yield {
+          createTestSubmission(testData.workspace, testData.methodConfigArrayType, testData.sset1, testData.userOwner,
+            Seq(testData.sset1), Map(testData.sset1 -> inputResolutionsList),
+            Seq.empty, Map.empty, st)
+        }
+      }.foreach { sub =>
+        runAndWait(submissionQuery.create(ctx, sub))
+      }
+    }
+
+    val result = runAndWait(workflowQuery.countWorkflowsByQueueStatus)
+    validateCountsByQueueStatus(result)
+    statusCounts.foreach { case (st, count) =>
+      result(st.toString) should be >= count
+    }
+  }
+
+  it should "count workflows by queue status by user" in withDefaultTestDatabase {
+    // Create a new test user and some test submissions
+    val testUserEmail = "testUser"
+    val testUserStatusCounts = Map(WorkflowStatuses.Submitted -> 1, WorkflowStatuses.Running -> 10, WorkflowStatuses.Aborting -> 100)
+    withWorkspaceContext(testData.workspace) { ctx =>
+      val testUser = RawlsUser(UserInfo(testUserEmail, OAuth2BearerToken("token"), 123, "123456789876543212346"))
```

Comment:
Isn't this already `testData.userWriter`? I think this user has already been created and saved.